pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/dictionary/DictionaryIndexTypeTest.java

```diff
@@ -224,27 +224,48 @@ public void newVarLength()
     }
 
     @Test
-    public void oldToNewConfConversion()
+    public void oldToNewConfConversionWithOnHeap()
         throws IOException {
-      _tableConfig.getIndexingConfig().setNoDictionaryColumns(
-          JsonUtils.stringToObject("[\"dimInt\"]", _stringListTypeRef)
-      );
       _tableConfig.getIndexingConfig()
           .setOnHeapDictionaryColumns(JsonUtils.stringToObject("[\"dimInt\"]", _stringListTypeRef));
+      convertToUpdatedFormat();
+      FieldConfig fieldConfig = getFieldConfigByColumn("dimInt");
+      assertNotNull(fieldConfig.getIndexes().get(new DictionaryIndexType().getPrettyName()));
```

Comment:
Shoudln't we check here that the DictionaryConfig is on heap here?

Also, I think it would be better to do not instantiate a new DIctionaryIndexType and instead use StandardIndexes.dictionary()

Reply:
Nice catch, I even missed the use var length check below.